frontend/src/components/UserProfile/UserProfileEditPassword.tsx

```diff
@@ -0,0 +1,253 @@
+import React, { useState } from 'react';
+import { useCookies } from 'react-cookie';
+import { useForm } from 'react-hook-form';
+import { connect } from 'react-redux';
+import { userUpdateProfileDetails, userUpdatePassword } from '../../actions';
+import { StoreState } from '../../reducers';
+
+import {
+  Alert,
+  Box,
+  Button,
+  CircularProgress,
+  IconButton,
+  InputAdornment,
+  Stack,
+  TextField,
+} from '@mui/material';
+import { AxiosError } from 'axios';
+import { Visibility, VisibilityOff } from '@mui/icons-material';
+
+export interface ProfileEditPasswordInput {
+  password: string;
+  new_password: string;
+  new_password_confirmation: string;
+}
+
+const formValidation = {
+  password: {
+    required: {
+      value: true,
+      message: 'This field is required.',
+    },
+    minLength: {
+      value: 6,
+      message: 'Password minimum length is 6',
+    },
+    maxLength: {
+      value: 255,
+      message: 'Full Name field max character up to 255 only.',
+    },
+  },
+  new_password: {
+    required: {
+      value: true,
+      message: 'This field is required.',
+    },
+    minLength: {
+      value: 6,
+      message: 'Password minimum length is 6',
+    },
+    maxLength: {
+      value: 255,
+      message: 'Email field max character up to 255 only.',
+    },
+  },
+  new_password_confirmation: {
+    required: {
+      value: true,
+      message: 'This field is required.',
+    },
+    minLength: {
+      value: 6,
+      message: 'Password minimum length is 6',
+    },
+    maxLength: {
+      value: 255,
+      message: 'Email field max character up to 255 only.',
+    },
+  },
+};
```

Comment:
separate this into a new json file

Reply:
done, inside index.tsx and imported as variable